OpenRA.Mods.Common/Traits/SupportPowers/SupportPowerManager.cs

```diff
@@ -256,6 +262,26 @@ public void Activate(Order order)
 			if (Info.OneShot)
 				PrerequisitesAvailable(false);
 		}
+
+		bool HasSufficientFunds(SupportPower power, bool activate = false)
```

Comment:
This activate flag isn't very nice, and is only used from one place.  Please cache the `PlayerResources` from `INotifyCreated` and then pull the `pr.TakeCash` out into the `Activate` method.